storage/flush.go

```diff
@@ -100,8 +103,14 @@ func (m *flushManager) Flush(curr time.Time) error {
 	m.setState(flushManagerFlushInProgress)
 	for _, ns := range namespaces {
 		// Flush first because we will only snapshot if there are no outstanding flushes
-		flushTimes := m.namespaceFlushTimes(ns, curr)
-		multiErr = multiErr.Add(m.flushNamespaceWithTimes(ns, flushTimes, flush))
+		flushTimes := m.namespaceFlushTimes(ns, tickStart)
+		shardBootstrapTimes, ok := dbBootstrapStates.namespaceBootstrapStates[ns.ID().String()]
```

Comment:
Hm, does `dbBootstrapStates.namespaceBootstrapStates` eventually get populated correctly?  Also should probably generate a map here so don't have to keep stringifying.  Although since it's not called that often maybe its ok.

Reply:
@r Not sure what you mean by populated correctly? Oh, do you mean in the case of a namespace change? Yes the dbBootstrapStates is generated in each tick of the mediator, so it would just happen next time around.

Yeah I thought about doing a genny map thing too, but its infrequent enough I thought it wasn't worth it (string() doesn't allocate I thought because string([]byte{}) doesn't allocate twice?

Reply:
and the map[string] itself is like 2-8 strings (depending on how many namespaces you have)

Reply:
It will allocate sadly, as the `[]byte(...)` underlying is mutable but strings are guarenteed immutable so it can't avoid an alloc.

That's fine though, yeah it's not really used much.